proof/courier.go

```diff
@@ -1287,6 +1293,57 @@ func FetchProofProvenance(ctx context.Context, originLocator Locator,
 	// is a reversal of that found in the proof file.
 	var reversedProofs []Blob
 	for {
+		// Before we attempt to fetch the proof from the potentially
+		// remote universe, we'll check our local archive to see if we
+		// already have it. This doesn't make sense in all cases, so we
+		// allow the local archive to be nil, in which case we just skip
+		// this step. The local universe might give us a single proof or
+		// a full file, depending on its type. If we get a full file, we
+		// are already done.
+		var (
+			fullProof Blob
```

Comment:
So this is'll be the _file_ or the state transition?

Reply:
Also why do we assume the local archive can only give us the full file? If the local archive is actually our own local universe, then it can receive the state transition blob.

Reply:
This will be the full proof file. The `FetchProof` method of the `Archiver` dictates that it has to be the full file, not just the state transition. So even if the given implementation is the local universe, it will assemble and return the full proof file.